packages/tinacms/src/auth/TinaCloudProvider.tsx

```diff
@@ -157,7 +157,9 @@ export const TinaCloudProvider = (
     [props.cms]
   )
   if (!cms.api.tina) {
-    cms.registerApi('tina', createClient(props))
+    cms.registerApi('tina', createClient({ ...props, branch: currentBranch }))
+  } else {
+    cms.api.tina.setBranch(currentBranch)
```

Comment:
This makes sense to me.  Would explain why `branch` wasn't working if we weren't instantiating the `client` with it.

I wonder if the CMS working properly is more a race condition than intentional coding?

Reply:
yeah, I was wondering about a race condition as well. I'll make this ready for review (James P has already tested and confirmed it fixed the issue for him)